core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -138,6 +138,10 @@ private[rest] class StandaloneSubmitRequestServlet(
     val driverExtraClassPath = sparkProperties.get("spark.driver.extraClassPath")
     val driverExtraLibraryPath = sparkProperties.get("spark.driver.extraLibraryPath")
     val superviseDriver = sparkProperties.get("spark.driver.supervise")
+    val masters = sparkProperties.get("spark.master")
```

Comment:
I also don't know this code, but it seems a bit funny that part of the code uses `masterUrl` and part uses the property `spark.master`. `masterUrl` seems to be a spark: URL constructed from the actual RPC endpoint. Is this difference going to cause problems? is it that `spark.master` changes and needs to override it? whatever the reason, maybe some comments would help.